src/java/org/apache/cassandra/metrics/DroppedMessageMetrics.java

```diff
@@ -44,6 +45,56 @@ public DroppedMessageMetrics(Verb verb)
+                                                         .put("BATCH_REMOVE_REQ", "BATCH_REMOVE")
+                                                         .put("BATCH_STORE_REQ", "BATCH_STORE")
+                                                         .put("COUNTER_MUTATION_REQ", "COUNTER_MUTATION")
+                                                         .put("HINT_REQ", "HINT")
+                                                         .put("MUTATION_REQ", "MUTATION")
+                                                         .put("RANGE_REQ", "RANGE_SLICE")
+                                                         .put("READ_REQ", "READ")
+                                                         .put("READ_REPAIR_REQ", "READ_REPAIR")
+                                                         .put("REQUEST_RSP", "REQUEST_RESPONSE")
+                                                         .build();
+
+        if (requestVerbsAlias.containsKey(currentScope))
+        {
+            dropped = Metrics.meter(factory.createMetricName("Dropped"),
+                                    new DroppedMessageMetrics.ScopeAliasMetricNameFactory()
+                                    .createMetricName("Dropped", requestVerbsAlias.get(currentScope)));
+            internalDroppedLatency = Metrics.timer(factory.createMetricName("InternalDroppedLatency"),
+                                                   new DroppedMessageMetrics.ScopeAliasMetricNameFactory()
+                                                   .createMetricName("InternalDroppedLatency", requestVerbsAlias.get(currentScope)));
+            crossNodeDroppedLatency = Metrics.timer(factory.createMetricName("CrossNodeDroppedLatency"),
+                                                    new DroppedMessageMetrics.ScopeAliasMetricNameFactory()
+                                                    .createMetricName("CrossNodeDroppedLatency", requestVerbsAlias.get(currentScope)));
+        }
+        else
+        {
+            dropped = Metrics.meter(factory.createMetricName("Dropped"));
+            internalDroppedLatency = Metrics.timer(factory.createMetricName("InternalDroppedLatency"));
+            crossNodeDroppedLatency = Metrics.timer(factory.createMetricName("CrossNodeDroppedLatency"));
+        }
+    }
+
+    static class ScopeAliasMetricNameFactory
+    {
+        public CassandraMetricsRegistry.MetricName createMetricName(String metricName, String aliasScope)
+        {
+            String groupName = CASClientWriteRequestMetrics.class.getPackage().getName();
+            String mbeanName = groupName + ':' +
+                               "type=" + "DroppedMessage" +
+                               ",scope=" + aliasScope +
+                               ",name=" + metricName;
+
+            return new CassandraMetricsRegistry.MetricName(groupName, "DroppedMessage", metricName, aliasScope, mbeanName);
+        }
```

Comment:
Could we not use `DefaultNameFactory.createMetricName` instead of creating a new Factory?